### PR TITLE
MAJOR: WCS reversal is non-functional

### PR DESCRIPTION
**Pull request by keflavich**
If you slice a cube with [::-1], it does not return a valid cube.  This is a
severe error that I don't really comprehend yet, so I'm starting with the
regression test.

### CODE REVIEW

**Comment by keflavich** [comment]
@astrofrog - does this ring any bells?  Somehow our tests didn't cover this, which is a HUGE oversight.

**Comment by keflavich** [comment]
I cancelled some of the travis builds since they were intentional failures

**Comment by keflavich** [comment]
several of the travis jobs have failed on code that is *not present* in the latest commit.  This appears to be a travis bug?  https://travis-ci.org/radio-astro-tools/spectral-cube/jobs/359491426#L1246

re-running travis...

**Comment by keflavich** [comment]
@astrofrog I think this turned out to be a very simple oversight, where I simply forgot to set one of the WCS parameters.  Could you just verify that I've done something halfway sensible, with the tests if nothing else?  This is an urgent bugfix, but I don't want to merge it without someone else's eyes on it.

File: spectral_cube/tests/test_spectral_cube.py [hunk 1]
```diff
@@ -940,6 +941,21 @@ def test_slice_wcs(view, naxis):
...
+    rcube = cube[view]
+    rrcube = rcube[view]
+
+    np.testing.assert_array_equal(np.diff(cube.spectral_axis),
+                                  -np.diff(rcube.spectral_axis))
+
+    np.testing.assert_array_equal(rrcube.spectral_axis.value,
+                                  cube.spectral_axis.value)
+    np.testing.assert_array_equal(rcube.spectral_axis.value,
+                                  cube.spectral_axis.value[::-1])
```
Comment (astrofrog) [inline spectral_cube/tests/test_spectral_cube.py]: Can you check that the world coordinate axes stay the same?

**Pull request merged**